tin_factory.go

```diff
@@ -0,0 +1,28 @@
+package gobltin
+
+// Factory for getting all validation functions
+func GetTinLookup(countryCode string) TinLookup {
+	switch {
+	case isEuropeanCountryCode(countryCode): // For the moment it only supports VIES lookup
+		return VIESLookup{}
+	// Add cases for other countries and their specific validators
+	default:
+		return nil // nil in case we don't have a validator for the country code
+	}
+}
+
+// List of all EU country codes supported by VIES
+var europeanCountryCodes = []string{
```

Comment:
I'd suggest trying to leverage as much as possible from GOBL for the different types. Instead of an array of `string`, I'd consider `[]l10n.Code` directly. You could also use the constants pre-defined in the GOBL `l10n` package.